src/magic_folder/participants.py

```diff
@@ -156,7 +157,7 @@ def mutable_dirnode(self, attribute, value):
         The Upload DMD must be a writable directory capability
         """
         uri = tahoe_uri_from_string(value)
-        if IDirnodeURI.providedBy(uri):
+        if IDirectoryURI.providedBy(uri):
             if not uri.is_readonly():
```

Comment:
I *think* this is implied by `IDirectoryURI`, though the documentation doesn't make it clear.

Reply:
hmmm, yeah reading the code I believe you're correct.

Tempted to leave it there so future-me doesn't have to re-read the Tahoe code every time ... maybe I should have put all this stuff in `capabilities.py` anyway? (e.g. `is_readonly_directory_cap()` and `is_readwrite_directory_cap()` approximately) ...

Reply:
or `is_directory_capability()` along with `is_readonly_capability()` or so ....

Reply:
took out redundant check at least